### PR TITLE
Adding AverageExtension postprocessor

### PR DESCRIPTION
**Pull request by jain651**
# Reason

Extension between two points in a numerical domain is required to compare modeling result to the experimentally observed results. Currently to calculate extension between two points in a domain, we need to obtain displacement in each direction (using PointValue postprocessor) for individual points and then manually obtain the deformation.

# Design

A new postprocessor is required that automatically gather the extension in each pair of points and average them to provide average extension between a pair to points.

# Impact

Reduces the length and better handling of input file and possible error in the manual calculations.

#251

### CODE REVIEW

**Comment by jain651** [comment]
@bwspenc this PR is ready for review

File: doc/content/source/postprocessor/AveragePercentExtension.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# AveragePercentExtension
+
+!syntax description /Postprocessors/AveragePercentExtension
+
+## Description
+
+This `PostProcessor` computes the average percentage extension between two sets of points specified in two arrays defined with the `first_point` and `last_point` parameters. The point positions account for the current displacements, so this distance will evolve over time for a deforming body. If the separation between two specific points is desired, single points rather than a arrays of points can be specified.
+
+This `PostProcessor` computes average percentage extension between a set of points specified in two array `first_point` and `last_point`.
```
Comment (bwspenc) [inline doc/content/source/postprocessor/AveragePercentExtension.md:9]: between two sets of points specified in two arrays:

Also, it's probably also worth stating that: The extensions between the points with matching positions in these arrays are computed, and then their average is computed. The lengths of these sets of points must be the same.

**Review by bwspenc — CHANGES_REQUESTED** [review]
Looks good -- just a minor change requested.

My only remaining concern on this is that I'm not sure whether this should compute the percentage. It seems more natural to me to just report delta L / L. Is that called "extension" or "extension ratio"? I'm not quite sure.  I always call it "delta L over L"! https://en.wikipedia.org/wiki/Deformation_(physics)

**Comment by bwspenc** [comment]
You were going to change the name to AverageExtensionRatio, right?

File: test/tests/AverageExtensionRatio/tests [hunk 2]
```diff
@@ -0,0 +1,24 @@
+[Tests]
+  issues = '#251'
+  design = 'AverageExtensionRatio.md'
+
+  [AverageExtensionRatio]
+    requirement = "AverageExtensionRatio computes the average extension ratio between two sets of points specified in two arrays: `first_point` and `last_point`. It should result in error when sizes of `first_point` and `last_point` do not match and size of `first_point` is zero "
```
Comment (bwspenc) [inline test/tests/AverageExtensionRatio/tests:6]: A few minor issues here:
1) There doesn't appear to be a test of the postprocessor under standard conditions.
2) If you're going to re-use the same -i file like you're doing for the error tests (which is a good idea), you should add `prereq` so they don't all run at the same time.
3) The way you have the requirements specified isn't really the best.  I think I wouldn't use `detail` in this case. I'd just have three tests with three requirements:
- The system shall provide a postprocessor that computes the average extension ratio between two sets of points.
- AverageExtensionRatio shall generate an error if the sizes of the specified sets of points do not match.
- AverageExtensionRatio shall generate an error if the size of either of the sets of points is zero.

**Review by bwspenc — CHANGES_REQUESTED** [review]
This looks good, just needs some minor work on the tests as noted.

**Review by bwspenc — APPROVED** [review]

**Comment by bwspenc** [comment]
By the way, you don't need to repeat the stuff in the issue in the pull request. Just say 'closes #251'.

**Pull request merged**